library/core/src/pin.rs

```diff
@@ -1755,13 +1755,18 @@ where
     }
 }
 
+/// Restricted to `Unpin` types, as it's not safe to obtain a mutable reference to a pinned value.
+///
+/// Note that because `Pin` is fundamental, it would normally be possible for crates to implement
+/// `DerefMut` for `Pin<T>` when `T` is a local type and not covered by this impl block. However,
+/// this impl block is written in a special way to prevent that for soundness reasons.
```

Comment:
```suggestion
/// For soundness reasons, implementations of `DerefMut` for `Pin<T>` are rejected even when `T` is
/// a local type not covered by this impl block. (Because `Pin` is [fundamental], such
/// implementations would normally be possible.)
///
/// [fundamental]: ../../reference/items/implementations.html#r-items.impl.trait.fundamental
```

Revised to put the bottom line upfront and to state it as a rule without speaking to its implementation (we could conceivably enforce this rule in other ways).  Also ,as @RalfJung often points out, "note that" tends to imply the rule was already stated and what follows is an implication, but that's not the case here.